internal/util/importutil/numpy_parser.go

```diff
@@ -343,6 +346,20 @@ func (p *NumpyParser) validateHeader(columnReader *NumpyColumnReader) error {
 				zap.String("fieldName", columnReader.fieldName))
 			return fmt.Errorf("illegal shape %d of numpy file for scalar field '%s', shape should be 1", shape, columnReader.fieldName)
 		}
+
+		// check varchar/JSON max length
+		if columnReader.dataType == schemapb.DataType_VarChar || columnReader.dataType == schemapb.DataType_JSON {
+			varcharLen, _, err := stringLen(columnReader.reader.GetDescType())
+			if err != nil {
+				log.Warn("Numpy adapter: failed to get max length of numpy file", zap.Error(err))
+				return err
+			}
+			if varcharLen > columnReader.maxVarcharLen {
+				log.Warn("Numpy adapter: length of varchar exceeds max length", zap.Int("len", varcharLen),
```

Comment:
Using "ength of varchar exceeds max length" here might not be appropriate, it is necessary to differentiate between JSON and VarChar.